radixdlt/src/main/java/com/radixdlt/consensus/EventCoordinator.java

```diff
@@ -83,13 +77,11 @@ private void processNewRound(Round round) {
 			return;
 		}
 
-		List<Atom> atoms = mempool.getAtoms(1, Sets.newHashSet());
-		if (!atoms.isEmpty()) {
-			QuorumCertificate highestQC = vertexStore.getHighestQC()
-				.orElseGet(this::makeGenesisQC);
+		Vertex proposal = proposalGenerator.generateProposal(this.pacemaker.getCurrentRound());
 
-			log.info("Starting round " + round + " with proposal " + atoms.get(0));
-			networkSender.broadcastProposal(new Vertex(highestQC, this.pacemaker.getCurrentRound(), atoms.get(0)));
+		// TODO: Handle empty proposals
```

Comment:
Wouldn't it be best to handle this now, rather than just letting the pipeline stall?

Reply:
Problem with handling it now is that because it is single node+responsiveness, rounds will just continue to increase as fast as it can making it harder to debug, so avoiding this change until later.